From 68852818f732b31d3deb7cc76ef16f4e61c41aec Mon Sep 17 00:00:00 2001
From: Colin Walters <walters@verbum.org>
Date: Sat, 9 May 2015 11:29:20 -0400
Subject: [PATCH] systemd: Set var-lib-nfs-rpc_pipefs.mount After= tmpfiles

OSTree is a mechanism for atomic updates of operating systems, with
designs for how system state is managed; in particular, `/var` should
start out empty, and components are responsible for creating content
there at runtime.

rpm-ostree consumes RPMs and commits them to an OSTree repository.
It has some support for automatically synthesizing systemd `tmpfiles.d`
snippets from RPM content in `/var` using systemd-tmpfiles.

However, in this case nfs-utils wants a mount point directory, and
it's running before systemd-tmpfiles. It should be perfectly fine to
do this mount after tmpfiles has run.

A better fix for this would be to move transient directories to
`/run`; However, that would be an invasive change, which can happen
after this fix.

Signed-off-by: Colin Walters <walters@verbum.org>
---
 systemd/var-lib-nfs-rpc_pipefs.mount | 1 +
 1 file changed, 1 insertion(+)

diff --git a/systemd/var-lib-nfs-rpc_pipefs.mount b/systemd/var-lib-nfs-rpc_pipefs.mount
index 33c5db6..26d1c76 100644
--- a/systemd/var-lib-nfs-rpc_pipefs.mount
+++ b/systemd/var-lib-nfs-rpc_pipefs.mount
@@ -1,6 +1,7 @@
 [Unit]
 Description=RPC Pipe File System
 DefaultDependencies=no
+After=systemd-tmpfiles-setup.service
 Conflicts=umount.target
 
 [Mount]
-- 
1.8.3.1

